services/metadata_service/tests/integration_tests/artifact_test.py

```diff
@@ -132,8 +131,12 @@ async def test_run_artifacts_get(cli, db):
     _first_artifact = (await add_artifact(db, flow_id=_task["flow_id"], run_number=_task["run_number"], step_name=_task["step_name"], task_id=_task["task_id"], artifact=ARTIFACT_A)).body
     _second_artifact = (await add_artifact(db, flow_id=_task["flow_id"], run_number=_task["run_number"], step_name=_task["step_name"], task_id=_task["task_id"], artifact=ARTIFACT_B)).body
 
+    # expect artifacts' tags to be overridden by tags of their ancestral run
+    update_objects_with_run_tags('artifact', [_first_artifact, _second_artifact], _run)
+
     # try to get all the created artifacts
-    await assert_api_get_response(cli, "/flows/{flow_id}/runs/{run_number}/artifacts".format(**_task), data=[_first_artifact, _second_artifact])
+    await assert_api_get_response(cli, "/flows/{flow_id}/runs/{run_number}/artifacts".format(**_task),
+                                  data=[_first_artifact, _second_artifact], data_is_unordered_list=True)
```

Comment:
When I ran the integration test suite locally, there was flakiness whenever we depend on returned object order (roughly fails 10-30% of the time as a suite).  We stop enforcing result ordering in all the places for `assert_api_get_response`.  New callers will still default to enforcing ordering to not surprise anyone here.